### ISSUE TITLE
Editor: header drafts count not updated when drafts are trashed

### ISSUE THREAD

**Issue by gravityrail** [Automattic/wp-calypso#12434]
#### Steps to reproduce
1. Start at any URL in Calypso, with more than one draft in the mini "drafts count" next to the Write button in the title bar.
2. Click the drafts count and click "See all Drafts"
3. Trash some drafts

#### What I expected

Drafts count in menu bar updates

#### What happened instead

Drafts count in menu bar does not update

#### Browser / OS version

Chrome 57 / OS 10.12.3

#### Screenshot / Video

<img width="1079" alt="drafts-not-updated" src="https://cloud.githubusercontent.com/assets/51896/24219381/cb27b886-0f03-11e7-8d9d-eb496a226802.png">

**Comment by mtias** [comment]
The blocker here is connecting "/posts" to state. It's still in older flux stores.

**Comment by jeherve** [comment]
Also in 790257-zen

Similar issues were also reported in the past in #9871 and #18125

**Comment by mtias** [comment]
@mattsherman if you are looking at an iteration on the posts page that would move to consume from state, this could be an easy win.

**Comment by mattsherman** [comment]
Thanks @mtias for putting this on Tanooki's radar. I'll look into where this might fit into our future iterations.

**Comment by markx2** [comment]
I have a similar issue

Drafts in Calypso is showing '1', but there is no draft in wp-admin.

I tried adding a new draft - the count went to 2, then when I trashed it it changed back to 1.

User reports this has been showing for some time.

2543539-hc

**Comment by catkit** [comment]
Another report here: 9361140-hc
No drafts, but Calypso drafts tab shows 4:
![download](https://user-images.githubusercontent.com/3521683/51047070-c8de1080-15ae-11e9-8db6-6f0e68a9cbd8.jpeg)

**Comment by thehenrybyrd** [comment]
Reproduced on my own test site.
In this case I deleted from within the block editor, using "Move to Trash" within Document > Status & Visibility. 
After trashing the post, I'm directed to the Trash tab, where I see two trashed posts, but the tab shows "1".

<img width="873" alt="Screen Shot 2019-06-17 at 10 32 23 AM" src="https://user-images.githubusercontent.com/15107387/59586622-014cc200-90ec-11e9-9fdc-0be9711d1b36.png">
